src/webgl/light.js

```diff
@@ -162,7 +162,12 @@ p5.prototype.directionalLight = function(v1, v2, v3, x, y, z) {
   var shader = this._renderer._useLightShader();
 
   //@TODO: check parameters number
-  var color = this.color.apply(this, [v1, v2, v3]);
+  var color;
```

Comment:
not sure how these changes relate to above. please separate into separate PR and tag with related issue.

Reply:
this change is necessary because the previous code is invalid when a `color` is passed to `directionalLight`/`pointLight` as the 1st parameter.

for example, if `c` is a `p5.Color` instance and you call `directionalLight(c, x, y, z)`, then the old code instantiates a color with (effectively) `color([c, x, y])` expecting `c,x,y` to be r,g,b values - but they're not.

this newer validation code catches this case and displays a warning, but during the unit test this causes a test failure. so, this change is necessary in order to get the tests to pass:

![image](https://user-images.githubusercontent.com/1088194/39399919-8c0dfab0-4adb-11e8-8f08-b58e45791cef.png)